lib/msf/core/exploit/remote/kerberos/service_authenticator/base.rb

```diff
@@ -279,6 +287,9 @@ def authenticate_via_kdc(options = {})
     realm = self.realm.upcase
     client_name = username
     server_name = "krbtgt/#{realm}"
+    offered_etypes = options.fetch(:offered_etypes) do
+      self.offered_etypes || Rex::Proto::Kerberos::Crypto::Encryption::DefaultOfferedEtypes
```

Comment:
if `self.offered_types` is an empty array it would be returned here right? is that possible/is that the intended outcome?